src/Controls/tests/UITests/Tests/AppiumPlatformsTestBase.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Reflection;
+using Microsoft.Maui.Appium;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using TestUtils.Appium.UITests;
+
+namespace Microsoft.Maui.AppiumTests
+{
+#if ANDROID
+	[TestFixture(TestDevice.Android)]
+#elif IOSUITEST
+	[TestFixture(TestDevice.iOS)]
+#elif MACUITEST
+	[TestFixture(TestDevice.Mac)]
+#elif WINTEST
+	[TestFixture(TestDevice.Windows)]
+#endif
+	public class AppiumPlatformsTestBase : AppiumUITestBase
+	{
+		TestDevice _testDevice;
+		public AppiumPlatformsTestBase(TestDevice device)
+		{
+			_testDevice = device;
+		}
+
+		[TearDown]
+		public void TearDown()
+		{
+			var testOutcome = TestContext.CurrentContext.Result.Outcome;
+			if (testOutcome == ResultState.Error ||
+				testOutcome == ResultState.Failure)
+			{
+				var logDir = (Path.GetDirectoryName(Environment.GetEnvironmentVariable("APPIUM_LOG_FILE")) ?? Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location))!;
+
+				var pageSource = Driver?.PageSource!;
+				File.WriteAllText(Path.Combine(logDir, $"{TestContext.CurrentContext.Test.MethodName}-PageSource.txt"), pageSource);
+
+				var screenshot = Driver?.GetScreenshot();
+				screenshot?.SaveAsFile(Path.Combine(logDir, $"{TestContext.CurrentContext.Test.MethodName}-ScreenShot.png"));
+			}
+
+			//this crashes on Android
+			if (!IsAndroid && !IsWindows)
+				Driver?.ResetApp();
+		}
+
+
+		[OneTimeSetUp]
+		public void OneTimeSetup()
+		{
+			InitialSetup();
+		}
+
+		[OneTimeTearDown()]
+		public void OneTimeTearDown()
+		{
+			Teardown();
+		}
+
+		public override TestConfig GetTestConfig()
+		{
+			var testConfig = new TestConfig(_testDevice, "com.microsoft.maui.uitests")
+			{
+				BundleId = "com.microsoft.maui.uitests",
+			};
+			switch (_testDevice)
+			{
+				case TestDevice.Android:
+					//_appiumOptions.AddAdditionalAppiumOption(AndroidMobileCapabilityType.AppPackage, "com.microsoft.maui.uitests");
+					// activity { com.microsoft.maui.uitests / crc64fa090d87c1ce7f0b.MainActivity}
+					//_appiumOptions.AddAdditionalAppiumOption(AndroidMobileCapabilityType.AppActivity, "MainActivity");
+					break;
+				case TestDevice.iOS:
+					testConfig.DeviceName = "iPhone X";
+					testConfig.PlatformVersion = Environment.GetEnvironmentVariable("IOS_PLATFORM_VERSION") ?? "14.4";
+					testConfig.Udid = Environment.GetEnvironmentVariable("IOS_SIMULATOR_UDID") ?? "";
+					break;
+				case TestDevice.Mac:
+
+					break;
+				case TestDevice.Windows:
+					testConfig.DeviceName = "WindowsPC";
+					testConfig.AppPath = Environment.GetEnvironmentVariable("WINDOWS_APP_PATH") ?? "";
```

Comment:
For this can we fill in the path for now? Since we only have the one test project?

```C#
					testConfig.AppPath = Environment.GetEnvironmentVariable("WINDOWS_APP_PATH") ??
#if DEBUG
						Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!, "..\\..\\..\\..\\..\\samples\\Controls.Sample.UITests\\bin\\Debug\\net7.0-windows10.0.20348\\win10-x64\\Controls.Sample.UITests.exe");
#else
						Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!, "..\\..\\..\\..\\..\\samples\\Controls.Sample.UITests\\bin\\Release\\net7.0-windows10.0.20348\\win10-x64\\Controls.Sample.UITests.exe");
#endif
					break;
```
or can we use the AppID of the project? That's how we did it with the control gallery
https://github.com/dotnet/maui/blob/a721d8babaab5caf108a1490f8c8fcb3fa2c2033/src/Compatibility/ControlGallery/test/WinUI.UITests/WindowsTestBase.cs#LL33C4-L33C101

Reply:
We can use the app ID - and you may have to with a packaged app. I think maybe only unpackaged will work with an exe?

Reply:
You should be able to launch the packaged app with the .exe as long as its already been installed

Reply:
The `exe` works fine

I wasn't able to get it to work with app ID though.

If I set the AppId on testconfig I just get

```
    OneTimeSetUp: OpenQA.Selenium.WebDriverArgumentException : Bad capabilities. Specify either app or appTopLevelWindow to create a session
```

looks like we set a few things in gallery

options.AddAdditionalCapability("appTopLevelWindow", topLevelWindowHandle);
options.AddAdditionalCapability("appArguments", "RunningAsUITests");

Reply:
I added the path lookup